### PR TITLE
feat: exponential moving average postprocessing

### PR DESCRIPTION
**Pull request by ab93**
- exponentially weighted moving average as an aggregator and as estimator
- fix unnecessary user warnings from trainer module
- raise error if multiple workers are passed in dataloader for StreamingDataset
- update quick-start notebook with univariate data (multivariate will be a separate notebook)

### CODE REVIEW

**Review by s0nicboOm — APPROVED** [review]

File: numalogic/postprocess.py [hunk 1]
```diff
@@ -9,24 +9,165 @@
...
+    n = len(arr)
+    theta = arr.reshape(1, -1)
+    powers = np.arange(1, n + 1)
+
+    # Calculate increasing powers of beta of the form [beta, beta**2, .., beta**n]
+    beta_powers = np.power(beta, powers)
+
+    # Calculate the reciprocals of beta powers
+    # [beta**(-1), beta**(-2), .., beta**(-n)]
+    beta_arr_inv = np.reciprocal(beta_powers.reshape(-1, 1))
```
Comment (qhuai) [inline numalogic/postprocess.py:68]: it'd be simpler to just do the following, and skip the reciprocal step.
```
powers = np.arange(n-1, 0, -1)
beta_powers = np.power(beta, powers)
```

File: numalogic/postprocess.py [hunk 2]
```diff
@@ -9,24 +9,165 @@
...
+
+    exp_avg = alpha * beta_powers[-1] * (theta @ beta_arr_inv)
+    if not bias_correction:
+        return exp_avg.item()
+
+    # Perform bias correction
+    corrected_exp_avg = exp_avg / (1.0 - np.power(beta, n))
```
Comment (qhuai) [inline numalogic/postprocess.py:75]: earlier, the code assured that beta > 1.0. however, we will be divide by zero exception if beta is 1.

File: numalogic/postprocess.py [hunk 3]
```diff
@@ -9,24 +9,165 @@
...
 from numalogic.tools import DataIndependentTransformers
+from numalogic.tools.exceptions import InvalidDataShapeError
+
+
+def _allow_only_single_feature(data: npt.NDArray[float]) -> None:
+    if data.ndim > 2:
+        raise InvalidDataShapeError(
+            f"Input data can only be 2 dimensions or less, input size: {data.shape}"
+        )
+    if data.ndim > 1 and data.shape[1] > 1:
```
[the remark this answers is not in the document]
Comment (ab93) [reply numalogic/postprocess.py:24]: As discussed, we also want to allow a rank zero array

File: numalogic/postprocess.py (hunk 2 above)
Comment (ab93) [reply numalogic/postprocess.py:75]: Good catch, fixing it.

File: numalogic/postprocess.py (hunk 1 above)
Comment (ab93) [reply numalogic/postprocess.py:68]: Acknowledged. Fixing it.

**Pull request merged**